gradle-baseline-java/src/test/java/com/palantir/baseline/BaselineExactDependenciesTest.java

```diff
@@ -0,0 +1,446 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline;
+
+import static com.palantir.gradle.testing.assertion.GradlePluginTestAssertions.assertThat;
+
+import com.palantir.gradle.testing.execution.GradleInvoker;
+import com.palantir.gradle.testing.execution.InvocationResult;
+import com.palantir.gradle.testing.junit.DisabledConfigurationCache;
+import com.palantir.gradle.testing.junit.GradlePluginTests;
+import com.palantir.gradle.testing.project.RootProject;
+import com.palantir.gradle.testing.project.SubProject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+@GradlePluginTests
+@DisabledConfigurationCache
+public class BaselineExactDependenciesTest {
+
+    private void applyStandardPlugins(RootProject rootProject) {
+        rootProject.buildGradle().plugins().add("java");
+        rootProject.buildGradle().plugins().add("com.palantir.baseline-exact-dependencies");
```

Comment:
diff from the old framework: https://github.com/palantir/gradle-baseline/pull/3396/files#diff-033634a655e9d3f2af7704b1da55c54fe2f62fa7060ef6ed16b1e8c1531b618aL34-L35 (previously it was adding 2 plugins without applying them - not really sure what is the reason for this though). 

For the test https://github.com/palantir/gradle-baseline/pull/3396/files#diff-b026a686feb05404b8f5d356e0e84b44b388981aa92e7af591723af9f435e84fR308 we would end up with a root build.gradle like this - which is invalid - 
```
plugins {
  ...
  id 'consistent-versions' apply false
  id 'consistent-versions'
}
```